app/src/main/java/org/onionshare/android/server/WebserverManager.kt

```diff
@@ -0,0 +1,116 @@
+package org.onionshare.android.server
+
+import android.util.Base64
+import android.util.Base64.NO_PADDING
+import android.util.Base64.URL_SAFE
+import com.mitchellbosecke.pebble.loader.ClasspathLoader
+import io.ktor.application.Application
+import io.ktor.application.ApplicationStarted
+import io.ktor.application.ApplicationStopped
+import io.ktor.application.call
+import io.ktor.application.install
+import io.ktor.features.CallLogging
+import io.ktor.features.StatusPages
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.content.resources
+import io.ktor.http.content.static
+import io.ktor.pebble.Pebble
+import io.ktor.pebble.PebbleContent
+import io.ktor.response.respond
+import io.ktor.routing.Route
+import io.ktor.routing.get
+import io.ktor.routing.routing
+import io.ktor.server.engine.ApplicationEngine
+import io.ktor.server.engine.embeddedServer
+import io.ktor.server.netty.Netty
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import java.security.SecureRandom
+import javax.inject.Inject
+
+internal const val PORT: Int = 17638
+
+class WebserverManager @Inject constructor() {
+
+    enum class State { STARTING, STARTED, STOPPING, STOPPED }
+
+    private val _state = MutableStateFlow(State.STOPPED)
+    val state: StateFlow<State> = _state
+
+    private val secureRandom = SecureRandom()
+    private var server: ApplicationEngine? = null
+
+    fun start(sendPage: SendPage) {
+        _state.value = State.STARTING
+        val staticPath = getStaticPath()
+        server = embeddedServer(Netty, PORT, watchPaths = emptyList()) {
+            install(CallLogging)
+            install(Pebble) {
+                loader(ClasspathLoader().apply { prefix = "assets/templates" })
+            }
+            installStatusPages(staticPath)
+            addListener()
+            routing {
+                defaultRoutes(staticPath)
+                sendRoutes(sendPage, staticPath)
+            }
+        }.also { it.start() }
+    }
+
+    fun stop() {
+        _state.value = State.STOPPING
+        server?.stop(1_000, 2_000)
+    }
+
+    private fun getStaticPath(): String {
+        val staticSuffixBytes = ByteArray(16).apply { secureRandom.nextBytes(this) }
+        val staticSuffix =
+            Base64.encodeToString(staticSuffixBytes, NO_PADDING or URL_SAFE).trimEnd()
```

Comment:
The python app encodes this as `Base32`, but I guess we don't need to pull this in for this. The randomized static path is probably not needed for as long as we don't support folders, but it also functions as a cache-breaker which is nice.